README.md

```diff
@@ -0,0 +1,168 @@
+The dog list can be filtered by breed using a controlled text input.
+
+Each dog has its own page to display some information fetched from the Wikipedia Media API. Information from the API may not be really related to the dog breed, as implementing this was beyond the scope of this project at the moment.
+When on the Pug's detail page (/dog/pug), a <Route> component is used to display a specific section.
+
+[Read more about the showdogs components](./components.md#showdogs)
+
+#### `<SubmitDog />`
+
+**At the moment this component is still under work and only showcases error handling**
+
+Allows the user to submit a dog image with the name and breed of the dog. The `<select>` element lists all the dogs' breeds received from the API and passed down by `<App />`.
+An image preview is shown next to the form after the user selects an image from their computer.
+Form submission calls an action that sends the form data to the corresponding endpoint, which currently don't exists and will be developed in the second part of this app. For now, it just showcases error handling.
+
+[Read more about SubmitDog](./components.md#submitDog)
+
+#### Error Handling
+
+To display alerts, warning, errors and any necessary type of message, the app uses `<MessageHandler />` and `<Snackbar />` components.
+
+The `<MessageHandler>` receives a list of messages which renders as `<Snackbars />` until they are either manually or automatically closed. If there are several messages at the same time they start to stack up from bottom to top. When the `<Snackbar />` component is closed, a callback function can be called to perform any other action, for example, removing the message from state.
+
+The `<Snackbar />` component shows either a string message or any given children. The component alone is able to autoclose after some time, defaults to 3000ms but can also be passed as a prop for different times. There are other options like not autoclosing, hiding the close button, and the style to be use for the message box (alert, warning, error).
+
+[Read more about the Error Handling components](./components.md#error-handling)
+
+### Disclaimer
+
+Some of the app features may be incomplete or can be enhanced. The focus of this project was not to have a polished off application, but to learn, comprehend and implement modern tools and processes.
+
+## Testing
+
+Testing was implemented on a late stage of the UI development, in parallel to Redux. All of Redux' actions and reducers were implemented using **Test Driven Development**. The `<SubmitDog />` component was tested after learning how to unit test on Redux. After this, the `<MessageHandler />` and `<Snackbar />` components were created using TDD.
+
+As testing was implemented late in the project, not all components have tests.
+
+To test the app in the project folder run `yarn test`.
```

Comment:
```suggestion
To test the app, in the project folder run `yarn test`.
```
But: To test the app, run `yarn test` in the root directory would be better

Reply:
🍤